### PR TITLE
Brush up

### PR DESCRIPTION
**Pull request by tatevikb**
Solution of 'Brush up your code' task.

### CODE REVIEW

File: coffee/src/CoffeeMachineSimulation.java [hunk 1]
```diff
@@ -0,0 +1,212 @@
+import java.util.Scanner;
+
+public class CoffeeMachineSimulation {
+
+    private enum State {
```
Comment (nsimonyan) [inline coffee/src/CoffeeMachineSimulation.java]: @tatevikb you can use separate enum . You can create an enum as you are creating class

File: coffee/src/CoffeeMachineSimulation.java [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+        for (int i = 0; i < input.length(); ++i) {
+            if (!Character.isDigit(input.charAt(i))) {
+                return false;
+            }
+        }
+
+        return true;
+    }
+
+    private void action(String input)
```
Comment (nsimonyan) [inline coffee/src/CoffeeMachineSimulation.java]: @tatevikb jan method is too big try to split it and also will be good to have comment on this one

**Review by nsimonyan — CHANGES_REQUESTED** [review]

File: coffee/src/CoffeeMachineSimulation.java [hunk 3]
```diff
@@ -0,0 +1,212 @@
...
+    {
+        boolean isInt = isInteger(input);
```
Comment (hhov88) [inline coffee/src/CoffeeMachineSimulation.java]: If isInt is mandatory just check it one time and print error massage, without checking it in all if cases

File: coffee/src/CoffeeMachineSimulation.java [hunk 4]
```diff
@@ -0,0 +1,212 @@
...
+        Off,
+        SelectCoffee,
+        FillWater,
+        FillCoffee,
+        FillMilk,
+        FillCups
+    }
+    private Scanner scan = new Scanner(System.in);
+
+    private final int ESPRESSO = 0;
```
Comment (hhov88) [inline coffee/src/CoffeeMachineSimulation.java]: It can be an enum too

File: coffee/src/CoffeeMachineSimulation.java [hunk 5]
```diff
@@ -0,0 +1,212 @@
...
+        System.out.println(prompt);
+        System.out.print("> ");
+    }
+
+    /**
+     *check to see if the input is integer representation
+     */
+    private boolean isInteger(String input)
+    {
+        for (int i = 0; i < input.length(); ++i) {
```
Comment (hhov88) [inline coffee/src/CoffeeMachineSimulation.java]: why just don't use a parseInt ?

File: coffee/src/CoffeeMachineSimulation.java (hunk 1 above)
Comment (tatevikb) [reply coffee/src/CoffeeMachineSimulation.java]: Separate enum is created. I would like to understand, what is advantage of it?

File: coffee/src/CoffeeMachineSimulation.java (hunk 2 above)
Comment (tatevikb) [reply coffee/src/CoffeeMachineSimulation.java]: Comments were added, method was spitted into 3 parts.

File: coffee/src/CoffeeMachineSimulation.java (hunk 3 above)
Comment (tatevikb) [reply coffee/src/CoffeeMachineSimulation.java]: Done

File: coffee/src/CoffeeMachineSimulation.java (hunk 4 above)
Comment (tatevikb) [reply coffee/src/CoffeeMachineSimulation.java]: I've created enum for coffee types and made the appropriate changes in the code.

File: coffee/src/CoffeeMachineSimulation.java (hunk 5 above)
Comment (tatevikb) [reply coffee/src/CoffeeMachineSimulation.java]: Done. Added try and catch statements.

**Review by nsimonyan — APPROVED** [review]

**Pull request merged**